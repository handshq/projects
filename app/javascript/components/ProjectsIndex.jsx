import React, { useEffect } from "react";
import axios from "axios";

export default function ProjectsIndex() {
  // useEffect(() => {
  //   axios.get("/projects.json").then((response) => {
  //     console.log(response.data)
  //   })
  // }, [])

  const projects = [
    { name: "Dummy 0" },
    { name: "Dummy 1" },
    { name: "Dummy 2" },
    { name: "Dummy 3" },
    { name: "Dummy 4" }
  ];

  return (
    <>
      <h1>Projects</h1>
      <div id="filters">

      </div>
      <table id="projects">
        <thead>
          <tr><th>Name</th></tr>
        </thead>
        <tbody>
          {projects.map((project, index) => {
            return (
              <tr key={index}><td>{project.name}</td></tr>
            )
          })}
        </tbody>
      </table>
    </>
  )
}
