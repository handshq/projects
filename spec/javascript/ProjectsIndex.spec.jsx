import React from "react";
import renderer from "react-test-renderer";
import sinon from "sinon";

import ProjectsIndex from "../../app/javascript/components/ProjectsIndex";

let server;

describe(ProjectsIndex, () => {
  beforeEach(() => {
    server = sinon.fakeServer.create();
  })

  afterEach(() => {
    server.restore()
  })

  it("renders correctly", () => {
    const tree = renderer.create(<ProjectsIndex />);

    expect(tree).toMatchSnapshot()
  })
})
