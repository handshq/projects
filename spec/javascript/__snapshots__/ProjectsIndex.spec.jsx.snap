// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsIndex renders correctly 1`] = `
[
  <h1>
    Projects
  </h1>,
  <div
    id="filters"
  />,
  <table
    id="projects"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Dummy 0
        </td>
      </tr>
      <tr>
        <td>
          Dummy 1
        </td>
      </tr>
      <tr>
        <td>
          Dummy 2
        </td>
      </tr>
      <tr>
        <td>
          Dummy 3
        </td>
      </tr>
      <tr>
        <td>
          Dummy 4
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
